metricbeat/module/docker/cpu/helper.go

```diff
@@ -78,21 +79,30 @@ func perCpuUsage(stats *dc.Stats) common.MapStr {
 // TODO: These helper should be merged with the cpu helper in system/cpu
 
 func totalUsage(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.CPUUsage.TotalUsage - stats.PreCPUStats.CPUUsage.TotalUsage)
+	return calculateLoad(stats.CPUStats.CPUUsage.TotalUsage, stats.PreCPUStats.CPUUsage.TotalUsage)
 }
 
 func usageInKernelmode(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.CPUUsage.UsageInKernelmode - stats.PreCPUStats.CPUUsage.UsageInKernelmode)
+	return calculateLoad(stats.CPUStats.CPUUsage.UsageInKernelmode, stats.PreCPUStats.CPUUsage.UsageInKernelmode)
 }
 
 func usageInUsermode(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.CPUUsage.UsageInUsermode - stats.PreCPUStats.CPUUsage.UsageInUsermode)
+	return calculateLoad(stats.CPUStats.CPUUsage.UsageInUsermode, stats.PreCPUStats.CPUUsage.UsageInUsermode)
 }
 
 func systemUsage(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.SystemCPUUsage - stats.PreCPUStats.SystemCPUUsage)
+	return calculateLoad(stats.CPUStats.SystemCPUUsage, stats.PreCPUStats.SystemCPUUsage)
 }
 
-func calculateLoad(value uint64) float64 {
-	return float64(value) / float64(1000000000)
+// This function is meant to calculate the % CPU time change between two successive readings.
+// The "oldValue" refers to the CPU statistics of the last read.
+// Time here is expressed by second and not by nanoseconde.
+// The main goal is to expose the %, in the same way, it's displayed by docker Client.
+func calculateLoad(newValue uint64, oldValue uint64) float64 {
+	value := float64(newValue) - float64(oldValue)
+	if value < 0 {
+		logp.Err("time change calculation failed")
```

Comment:
SGTM: I would add the two values at the end or as part of the log message something like `new stats value (%v)  is lower then the old one (%v)`